csharp/src/Drivers/Apache/Thrift/Service/Rpc/Thrift/TBinaryColumn.cs

```diff
@@ -83,14 +84,14 @@ public TBinaryColumn DeepCopy()
 
                   values = new ArrowBuffer.Builder<byte>();
                   int offset = 0;
-                  offsetBuffer = new byte[(length + 1) * 4];
+                  offsetBuffer = new byte[(length + 1) * IntSize];
                   var memory = offsetBuffer.AsMemory();
                   var typedMemory = Unsafe.As<Memory<byte>, Memory<int>>(ref memory).Slice(0, length + 1);
 
                   for(int _i197 = 0; _i197 < length; ++_i197)
                   {
                     //typedMemory.Span[_i197] = offset;
-                    StreamExtensions.WriteInt32LittleEndian(offset, memory.Span, _i197 * 4);
+                    StreamExtensions.WriteInt32LittleEndian(offset, memory.Span, _i197 * IntSize);
```

Comment:
For .NET 6+ we could use e.g. BinaryPrimitives.WriteInt32LittleEndian for what I suspect is an efficiency gain. This could be done as a followup and/or performance-tested.

Reply:
Okay will create a follow-up to .NET 6+ support for `BinaryPrimitives.WriteInt32LittleEndian`